exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/aggregate/HashAggTemplate.java

```diff
@@ -325,10 +325,13 @@ public AggOutcome doWork() {
                 if (EXTRA_DEBUG_1) {
                   logger.debug("Received new schema.  Batch has {} records.", incoming.getRecordCount());
                 }
-                newSchema = true;
-                this.cleanup();
-                // TODO: new schema case needs to be handled appropriately
-                return AggOutcome.UPDATE_AGGREGATOR;
+                final BatchSchema newIncomingSchema = incoming.getSchema();
+                if ((! newIncomingSchema.equals(schema)) && schema != null) {
```

Comment:
You shouldn't put a hack in to ignore a valid schema change. I don't know all the details but if this is related to HBase, my thought is you probably need to resolve 4010 to avoid spurious schema changes. You should make sure to avoid propagating schema changes that are not real/required. Solving on a particular operator just means if the query plan changes, you'll probably get an error in a different operator.